### PR TITLE
Validate FxOS Add-ons version number at upload (bug 1204746)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1204746

### CODE REVIEW

**Comment by diox** [comment]
Contains a refactor removing `ExtensionParser` since `ExtensionValidator` kinda made it obsolete.

File: mkt/extensions/models.py [hunk 1]
```diff
@@ -169,17 +203,9 @@ def update_manifest_fields_from_latest_public_version(self):
...
-            parsed_data = ExtensionParser(
-                None, manifest_contents=version.manifest).parse()
-        except ValidationError:
-            # This should not happen, the manifest should be valid, but there
-            # is not much we can do from this method.
-            return
-        data = {k: parsed_data[k] for k in fields if k in parsed_data}
+        # We need to re-extract the fields from manifest contents because some
+        # fields like default_language are transformed before being stored.
+        data = self.__class__.extract_manifest_fields(version.manifest)
```
Comment (washort) [inline mkt/extensions/models.py]: hmm maybe `extract_manifest_fields` should just be a function?

File: mkt/extensions/models.py [hunk 2]
```diff
@@ -401,6 +436,13 @@ def unsigned_download_url(self):
         return absolutify(
             reverse('extension.download_unsigned', kwargs=kwargs))
 
+    @classmethod
+    def version_compare(cls, version1, version2):
```
Comment (washort) [inline mkt/extensions/models.py]: maybe name this `version_lt` to make it clear what the operation is

**Comment by washort** [comment]
r+wc

**Pull request merged**